pkg/translator/sumoicmarshaler/sumo_ic_marshaler.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumoicmarshaler
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type SumoICMarshaler struct {
+	format string
+}
+
+func (marshaler *SumoICMarshaler) Format() string {
+	return marshaler.format
+}
+
+func NewSumoICMarshaler() SumoICMarshaler {
+	return SumoICMarshaler{}
+}
+
+func logEntry(buf *bytes.Buffer, format string, a ...interface{}) {
+	buf.WriteString(fmt.Sprintf(format, a...))
+	buf.WriteString("\n")
+}
+
+func attributeValueToString(v pcommon.Value) string {
+	switch v.Type() {
+	case pcommon.ValueTypeString:
+		return v.StringVal()
+	case pcommon.ValueTypeBool:
+		return strconv.FormatBool(v.BoolVal())
+	case pcommon.ValueTypeDouble:
+		return strconv.FormatFloat(v.DoubleVal(), 'f', -1, 64)
+	case pcommon.ValueTypeInt:
+		return strconv.FormatInt(v.IntVal(), 10)
+	case pcommon.ValueTypeSlice:
+		return sliceToString(v.SliceVal())
+	case pcommon.ValueTypeMap:
+		return mapToString(v.MapVal())
+	default:
+		return fmt.Sprintf("<Unknown OpenTelemetry attribute value type %q>", v.Type())
+	}
+}
+
+func sliceToString(s pcommon.Slice) string {
+	var b strings.Builder
+	b.WriteByte('[')
+	for i := 0; i < s.Len(); i++ {
+		if i < s.Len()-1 {
+			fmt.Fprintf(&b, "%s, ", attributeValueToString(s.At(i)))
+		} else {
+			b.WriteString(attributeValueToString(s.At(i)))
+		}
+	}
+
+	b.WriteByte(']')
+	return b.String()
+}
+
+func mapToString(m pcommon.Map) string {
+	var b strings.Builder
+	b.WriteString("{\n")
+
+	m.Sort().Range(func(k string, v pcommon.Value) bool {
+		fmt.Fprintf(&b, "     -> %s: %s(%s)\n", k, v.Type(), v.AsString())
+		return true
+	})
```

Comment:
Could you add UT for this code? Does IC really marshals maps that way?

Reply:
I believe we should use `json` format here

Reply:
As we can use AsString() function to stringify attributeValue above function is no longer needed